### ISSUE TITLE
switch_case_on_newline false positive on Xcode 12 beta

### ISSUE THREAD

**Issue by kacper1703** [realm/SwiftLint#3253]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

On Xcode 12.0 beta the `switch_case_on_newline` is triggered on a `catch` in the examples provided below:

```swift
do {
  let loadedToken = try tokenManager.decodeToken(from: response)
  return loadedToken
} catch { throw error } // switch_case_on_newline violation here at `catch`
```

```swift
  do {
    let components = try JWTComponents(token: token, authorizationKey: authorizationKey)
    XCTAssertEqual(components.payload.userId, "1234567890")
  } catch { XCTFail(error.localizedDescription) } // switch_case_on_newline violation here at `catch`
```

### Environment

* SwiftLint version: **0.39.2**
* Installation method: **CocoaPods**
* Xcode **12.0 beta (12A6159)**
* Configuration file:

```yml
disabled_rules:
  - type_name
  - vertical_parameter_alignment
  - function_default_parameter_at_end

opt_in_rules:
- array_init
- closure_end_indentation
- closure_parameter_position
- closure_spacing
- collection_alignment
- contains_over_filter_count
- contains_over_filter_is_empty
- contains_over_first_not_nil
- convenience_type
- discouraged_object_literal
- duplicate_enum_cases
- empty_collection_literal
- empty_count
- empty_string
- empty_xctest_method
- expiring_todo
- fallthrough
- fatal_error_message
- first_where
- for_where
- force_unwrapping
- implicit_return
- identical_operands
- iteral_expression_end_indentation
- last_where
- legacy_multiple
- legacy_random
- lower_acl_than_parent
- modifier_order
- number_separator
- operator_usage_whitespace
- overridden_super_call
- private_outlet
- private_action
- prohibited_super_call
- sorted_first_last
- sorted_imports
- static_operator
- strong_iboutlet
- switch_case_on_newline
- toggle_bool
- trailing_closure
- unavailable_function
- unneeded_parentheses_in_closure_argument
- unowned_variable_capture
- untyped_error_in_catch
- unused_declaration
- vertical_parameter_alignment_on_call
- xct_specific_matcher
- yoda_condition
```

**Comment by kacper1703** [comment]
Could anyone check on that before Xcode 12 is officially released?
